Sources/Receipt/ReceiptDetailSectionData.swift

```diff
@@ -0,0 +1,115 @@
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import HyperwalletSDK
+
+enum ReceiptDetailSectionHeader: String {
+    case transaction, details, fee, notes
+}
+
+protocol ReceiptDetailSectionData {
+    var receiptDetailSectionHeader: ReceiptDetailSectionHeader { get }
+    var rowCount: Int { get }
+    var title: String { get }
+    var cellIdentifier: String { get }
+}
+
+extension ReceiptDetailSectionData {
+    var rowCount: Int { return 1 }
+    var title: String { return "receipt_details_section_header_\(receiptDetailSectionHeader.rawValue)".localized() }
+}
+
+struct ReceiptDetailSectionTransactionData: ReceiptDetailSectionData {
+    var receiptDetailSectionHeader: ReceiptDetailSectionHeader { return .transaction }
+    var cellIdentifier: String { return ReceiptTransactionTableViewCell.reuseIdentifier }
+    let tableViewCellConfiguration: ReceiptTransactionCellConfiguration
+
+    init(from receipt: HyperwalletReceipt) {
+        tableViewCellConfiguration = ReceiptTransactionCellConfiguration(
+            type: receipt.type.rawValue.lowercased().localized(),
+            entry: receipt.entry.rawValue,
+            amount: receipt.amount,
+            currency: receipt.currency,
+            createdOn: ISO8601DateFormatter
+                .ignoreTimeZone
+                .date(from: receipt.createdOn)!
+                .format(for: .date),
+            iconFont: HyperwalletIcon.of(receipt.entry.rawValue).rawValue)
+    }
+}
+
+struct ReceiptDetailSectionDetailData: ReceiptDetailSectionData {
+    var rows: [(title: String, value: String)] = []
+    var receiptDetailSectionHeader: ReceiptDetailSectionHeader { return .details }
+    var rowCount: Int { return rows.count }
+    var cellIdentifier: String { return ReceiptDetailTableViewCell.reuseIdentifier }
+
+    init(from receipt: HyperwalletReceipt) {
+        let receiptId = receipt.journalId
+        rows.append((title: "receipt_details_receipt_id".localized(), value: receiptId))
+
+        let dateTime = ISO8601DateFormatter.ignoreTimeZone.date(from: receipt.createdOn)!.format(for: .dateTime)
+        rows.append((title: "receipt_details_date".localized(), value: dateTime))
+
+        if let charityName = receipt.details?.charityName {
+            rows.append((title: "receipt_details_charity_name".localized(), value: charityName))
+        }
+        if let checkNumber = receipt.details?.checkNumber {
+            rows.append((title: "receipt_details_check_number".localized(), value: checkNumber))
+        }
+        if let clientPaymentId = receipt.details?.clientPaymentId {
```

Comment:
Not sure if we should truncate client transaction ID if it's too long. Need to confirm with Jaspreet or Willem

Reply:
In this case, you can rotate the iPhone to the landscape.